### DISCUSSION TITLE
Application built

### DISCUSSION THREAD

**Discussion by omodia** [mienaiyami/yomikiru#162] [Q&A] [answered]
What does the app use for,
1. Backend
2. Frontend (UI)
3. Framework or library

**Comment by mienaiyami** [comment, marked as answer]
There is no backend, but you can say its electronJS. As for frontend its reactJS. Everything is written in typescript.

**Reply by omodia** [reply]
> There is no backend, but you can say its electronJS.

- I heard electron has critical perfomane issues, is it true?
- Is that one of the reasons you mentioned about building the same app in tauri previously?

**Reply by mienaiyami** [reply]
> I heard electron has critical perfomane issues, is it true?

The "critical" is not true now, it was many years ago. Though there is still little issues.

> Is that one of the reasons you mentioned about building the same app in tauri previously?

To me there is no visible performance difference, only difference is in RAM usage and app size. 
As it uses webview, there is not much control over app. Also its new and lacks features and uses rust as backend, which is not easy to use.
Also the ecosystem, it is overall very hard to use as of now, for big and complex apps.

**Comment by omodia** [comment]
> As for frontend its reactJS.

Have you heard of [Svelte?](https://svelte.dev/)

**Reply by mienaiyami** [reply]
> Have you heard of [Svelte?](https://svelte.dev/)

It is not as easy as picking it and using it. I started working on this app before svelte was that famous and react was the first such library i used. If you go way back, I just had pure JS and no libraries. 
Moreover, I dont think its possible for me to change from react to svelte (even though i have used svelte before) because it will take too much time and it is impractical for me to work on it as I get literally nothing out of it, I have no profit and dont have a job yet, Im just a student. I fix issues and add features because thats what I would want in this app as well. As i mentioned above, it was just a simple pure JS app first which ran in browser then moved to electron because of file system needs. I had just made it for myself, didnt know how to use git/github and was very beginner in development. At that time, I didnt had much experience and options and I never expected to even release this app, it was on a friend's request to use app as well.

https://gist.github.com/tkrotoff/b1caa4c3a185629299ec234d2314e190

**Reply by omodia** [reply]
> I have no profit and dont have a job yet, Im just a student.

With your experience and expertise, you will surely get a job.

> I fix issues and add features because thats what I would want in this app as well.

I am thankful that you chose to upload this app on github and added the features that I requested.
Once again, Thank you.
